packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1138,6 +1139,24 @@ export interface BuildEnvironment {
   readonly environmentVariables?: { [name: string]: BuildEnvironmentVariable };
 }
 
+/** Optional arguments to {@link IBuildImage.binder} - currently empty. */
+export interface BuildImageBindOptions {}
```

Comment:
I think this is fine to not put in. Adding optional arguments is not a breaking change.

Reply:
You cannot have optional arguments on interface declarations in TypeScript.

Reply:
You can have `arg: Type | undefined` arguments which jsii will treat the same.

Reply:
Is that a jsii restriction? Seems to work fine:

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgEJxBuyDeAoZQ5COMACjDgGcxgAbATwH4AuZAIwHtO6SQBKNgDdOwACYBuPAF8gA

Reply:
You're right, apologies. I was thinking about default arguments, not optional arguments.

But is adding an optional argument later really a backwards-compatible change? If I have

```ts
export interface IBibdableBuildImage extends IBuildImage {
  bind(scope: Construct, project: IProject): BuildImageConfig;
}
```

and I later change it to

```ts
export interface IBibdableBuildImage extends IBuildImage {
  bind(scope: Construct, project: IProject, options? BuildImageBindOptions): BuildImageConfig;
}
```

Won't that break, for example, Java implementations of `IBibdableBuildImage`?

Reply:
Hm. You might be right. Ask Romain, I guess :)

Reply:
So this won't break because:
- In **Java** we generate overload signatures by shaving the last optional argument until the last argument is no longer optional
- In **C#** we'll generate a signature with a default value (of `null`) for this argument, which reproduces the JS behavior
- In **Python** we'll generate a signature with a default value (of `None`) for this argument, which reproduces the JS behavior

Reply:
Oh wait this is an interface... I'm not sure we generate default overrides on those (we could totally do it if we don't already).

Reply:
That needs to be a jsii-diff issue then :(